### PR TITLE
jpexs: 18.4.1 -> 20.1.0

### PR DESCRIPTION
**Pull request by Semnodime**
## Description of changes
Update nixpkgs.jpexs to version 20.1.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of `ffdec` java application
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by donovanglover — COMMENTED** [review]
Welcome to nixpkgs! The checklist indicates that you haven't tested this package, but it's a major version bump. Make sure to go through the [changelog](https://github.com/jindrapetrik/jpexs-decompiler/releases) for any changes that might need to be done to the package.

The commit message and title of this PR should be changed to `jpexs: 18.4.1 -> 20.1.0`

**Comment by Semnodime** [comment]
It's a java application which does not rely on any external functionality AFAIK, so the nixpkg should not require any change besides the version increment and hash update.
TBH, I don't know how to test functionality of the nixpkg. Where can I read up on that? I just know how to run applications using `nix-env -iA nixos.jpexs` as instructed by https://search.nixos.org/packages.

**Comment by donovanglover** [comment]
If you have flakes enabled you can use `nix run .#jpexs` in your nixpkgs fork to test it out. You can also use `nixpkgs-review pr --print-result 331554` and paste the output here, which would also let you test the binary in `./result`.

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 331554` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>jpexs</li>
  </ul>
</details>

**Comment by Semnodime** [comment]
Running `nix run 'nixpkgs#nixpkgs-review' pr 331554` on x86_64-linux apparently failes to succeed due to a hash mismatch (`error: hash mismatch in fixed-output derivation`):
Which reading do you recommend to aquire the skills / knowledge required to further debug and eventually fix this?
```
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/331554/head:refs/nixpkgs-review/1
remote: Enumerating objects: 2286, done.
remote: Counting objects: 100% (1844/1844), done.
remote: Compressing objects: 100% (713/713), done.
remote: Total 2286 (delta 1406), reused 1414 (delta 1083), pack-reused 442
Receiving objects: 100% (2286/2286), 2.10 MiB | 6.69 MiB/s, done.
Resolving deltas: 100% (1473/1473), completed with 401 local objects.
From https://github.com/NixOS/nixpkgs
 * [new branch]                master                -> refs/nixpkgs-review/0
 * [new ref]                   refs/pull/331554/head -> refs/nixpkgs-review/1
$ git worktree add /home/user/.cache/nixpkgs-review/pr-331554/nixpkgs 8764d898c4f365d98ef77af140b32c6396eb4e02
Preparing worktree (detached HEAD 8764d898c4f3)
Updating files: 100% (42621/42621), done.
HEAD is now at 8764d898c4f3 Merge pull request #330741 from emilazy/push-rlvumrtyxrmk
$ git merge --no-commit --no-ff 51236af4196d1abcc16ac13818f8569625a86409
Automatic merge went well; stopped before committing as requested
warning: ignoring the client-specified setting 'system', because it is a restricted setting and you are not a trusted user
$ nix build --file /nix/store/c5ym3hahib1jj9da18niib26a1ly68f7-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/user/.cache/nixpkgs-review/pr-331554/nixpkgs nixpkgs-overlays=/tmp/tmpq5nyas6g' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr system x86_64-linux --argstr nixpkgs-path /home/user/.cache/nixpkgs-review/pr-331554/nixpkgs --argstr nixpkgs-config-path /tmp/tmpgdm6u65e.nix --argstr attrs-path /home/user/.cache/nixpkgs-review/pr-331554/attrs.nix
warning: ignoring the client-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
error: hash mismatch in fixed-output derivation '/nix/store/zghivikk0mm4scd9lp7mdf400kfyiplp-source.drv':
         specified: sha256-fLG29PsLJkuCMu0kXHWdFg0Lnvw79O1nQKNVqw2PJBA=
            got:    sha256-ytGtylhyNSdKfuPclZRJasOb/cskW65hMd4NM/q+/Ko=
error: 1 dependencies of derivation '/nix/store/j15md9i1axly3a7wvl4va08c65387lmn-jpexs-20.1.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xpnkv3m7xzghi4p7xz86a2x1yyyl26i3-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/331554

1 package failed to build:
jpexs

error: build log of '/nix/store/j15md9i1axly3a7wvl4va08c65387lmn-jpexs-20.1.0.drv^*' is not available
error: build log of '/nix/store/09a1lbv292xpmjzaj88k0sffbfq5kig5-jpexs-20.1.0' is not available
$ /nix/store/pf3c5fg5jb2mfa3ysmayk4xv2r9wb8iw-nix-2.18.5/bin/nix-shell --argstr system x86_64-linux --argstr nixpkgs-path /home/user/.cache/nixpkgs-review/pr-331554/nixpkgs --argstr nixpkgs-config-path /tmp/tmpgdm6u65e.nix --argstr attrs-path /home/user/.cache/nixpkgs-review/pr-331554/attrs.nix --nix-path 'nixpkgs=/home/user/.cache/nixpkgs-review/pr-331554/nixpkgs nixpkgs-overlays=/tmp/tmpq5nyas6g' /nix/store/c5ym3hahib1jj9da18niib26a1ly68f7-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix
```

File: pkgs/development/tools/jpexs/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 stdenv.mkDerivation rec {
   pname = "jpexs";
-  version = "18.4.1";
+  version = "20.1.0";
 
   src = fetchzip {
     url = "https://github.com/jindrapetrik/jpexs-decompiler/releases/download/version${version}/ffdec_${version}.zip";
-    sha256 = "sha256-aaEL3xJZkFw78zo3IyauWJM9kOo0rJTUSKmWsv9xQZ8=";
+    sha256 = "sha256-fLG29PsLJkuCMu0kXHWdFg0Lnvw79O1nQKNVqw2PJBA=";
```
Comment (donovanglover) [inline pkgs/development/tools/jpexs/default.nix:9]:
```suggestion
    hash = "sha256-ytGtylhyNSdKfuPclZRJasOb/cskW65hMd4NM/q+/Ko=";
```
Try changing the hash here from the `specified` one in the error to the `got` one.

The `sha256` -> `hash` change is unrelated (we're trying to replace `sha256` with `hash` in nixpkgs)

**Comment by Semnodime** [comment]
```
1 package built:
jpexs
```

Also, `nix run '.#jpexs'` works flawlessly.

**Comment by donovanglover** [comment]
LGTM after removing the `v` in the commit message. Should be `jpexs: 18.4.1 -> 20.1.0` instead of `v20.1.0`

**Review by donovanglover — APPROVED** [review]
LGTM. Grats on your first contribution to nixpkgs!

**Comment by Semnodime** [comment]
When will the new version of ffdec (jpexs) show up when running `nix-env -iA nixpkgs.jpexs` ?

**Comment by donovanglover** [comment]
You can track the progress of your pull request here: https://nixpkgs-tracker.ocfox.me/?pr=331554

You'll have access to the new version if you're following `nixos-unstable` and update your system when the branch name changes from red to green.
